### DISCUSSION TITLE
Can't log into TEAM

### DISCUSSION THREAD

**Discussion by cemendes** [aws-samples/iam-identity-center-team#138] [Q&A]
I did a new install of TEAM in a child account (Team Account). All went well, including the execution of the cognito.sh script(no errors) and now after the login, I see the TEAM page loading non-stop. 
![2024-01-06_23-00-27 (1)](https://github.com/aws-samples/iam-identity-center-team/assets/16995556/72f2edf7-cbf5-4d02-a0e6-e1fe1e87909b)

In the team account, I checked cloud trail and I see this error when trying to log-in. 

Any ideas would be appreciated. 

Field | Value
-- | --
1704602118841
730335522437:/aws/lambda/teamgetGroups-main
2024/01/07/[$LATEST]b44cc22778ae429eb5c11da269e60a62
**[ERROR]** ParamValidationError: Parameter validation failed:Invalid type for parameter MemberId.UserId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>Traceback (most recent call last):  File "/var/task/index.py", line 135, in handler    groupData = list_idc_group_membership(userId)  File "/var/task/index.py", line 116, in list_idc_group_membership    for page in paginator:  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 269, in __iter__    response = self._make_request(current_kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 357, in _make_request    return self._method(**current_kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call    return self._make_api_call(operation_name, kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 919, in _make_api_call    request_dict = self._convert_to_request_dict(  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 990, in _convert_to_request_dict    request_dict = self._serializer.serialize_to_request(  File "/opt/python/lib/python3.9/site-packages/botocore/validate.py", line 381, in serialize_to_request    raise ParamValidationError(report=report.generate_report())
1704602111105

**Comment by GeoDia** [comment]
Same problem here. @cemendes did you manage to solve it? Any advice would be highly appreciated!

**Comment by tawoyinfa** [comment]
@GeoDia  this issue has been fixed in recent updates #151. Is this a new or existing deployment ?

**Reply by GeoDia** [reply]
@tawoyinfa Thank you for the quick response. This is a new deployment, I tried once on 28 Jan and once today after running destroy.sh and cleaned up everything as the documentation states. I see the following error in my CloudTrail logs which seems related:

```
"eventTime": "2024-02-02T15:36:02Z",
    "eventSource": "identitystore.amazonaws.com",
    "eventName": "GetUserId",
    "awsRegion": "eu-west-1",
    "sourceIPAddress": "34.254.189.180",
    "userAgent": "Boto3/1.26.85 Python/3.9.18 Linux/5.10.201-213.748.amzn2.x86_64 exec-env/AWS_Lambda_python3.9 Botocore/1.29.85",
    "errorCode": "ResourceNotFoundException",
    "errorMessage": "USER not found.",
    "requestParameters": {
        "identityStoreId": "d-9367427037",
        "alternateIdentifier": {
            "uniqueAttribute": {
                "attributePath": "userName",
                "attributeValue": {
                    "untyped": {
                        "object": "HIDDEN_DUE_TO_SECURITY_REASONS"
                    }
                }
            }
        }
    }
```

To give some more context: I deploy TEAM in a test account, I have IDC in management account. I have created two users manually in IDC with which I try to log in. I successfully log in to IDC dashboard, then click TEAM and it hangs at loading. 

I run update.sh to be sure but I get: "No changes to deploy. Stack TEAM-IDC-APP is up to date".

**Comment by tawoyinfa** [comment]
@GeoDia  try redploying it again and let me know if the issue persists

**Reply by GeoDia** [reply]
@tawoyinfa I just redeployed and I get the same result (screen hanging there). Init, deploy and cognito scripts run without errors.

Feel free to ask me for any logs or anything else to resolve this. 
Thanks!

**Comment by GeoDia** [comment]
@tawoyinfa Checked CloudWatch logs as well and found the below in case it helps:
```
[ERROR] ParamValidationError: Parameter validation failed:
Invalid type for parameter MemberId.UserId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
Traceback (most recent call last):
  File "/var/task/index.py", line 135, in handler
    groupData = list_idc_group_membership(userId)
  File "/var/task/index.py", line 116, in list_idc_group_membership
    for page in paginator:
  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 269, in __iter__
    response = self._make_request(current_kwargs)
  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 357, in _make_request
    return self._method(**current_kwargs)
  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 919, in _make_api_call
    request_dict = self._convert_to_request_dict(
  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 990, in _convert_to_request_dict
    request_dict = self._serializer.serialize_to_request(
  File "/opt/python/lib/python3.9/site-packages/botocore/validate.py", line 381, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
```

**Comment by matty-carroll-krakenflex** [comment]
Hello, I'm also seeing the same issue. I haven't seen the above error but do have this one repeating in the /aws/lambda/teamgetGroups-main log group

`[ERROR] IndexError: list index out of range
Traceback (most recent call last):
  File "/var/lang/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/var/task/index.py", line 69, in <module>
    sso_instance = get_identity_store_id()
  File "/var/task/index.py", line 64, in get_identity_store_id
    return response['Instances'][0]['IdentityStoreId']`

**Reply by matty-carroll-krakenflex** [reply]
Update - managed to resolve this by deploying TEAM into the same region as Identity Centre (eu-west-1) rather than the default us-east-1

**Comment by GeoDia** [comment]
Solved the problem by creating a new AWS account only for deploying TEAM.
